languages/go/oso.go

```diff
@@ -152,6 +157,42 @@ func (o Oso) IsAllowed(actor interface{}, action interface{}, resource interface
 	}
 }
 
+/*
+Return a set of actions allowed by the given (actor, resource) combination allowed
+by the policy.
+*/
+func (o Oso) GetAllowedActions(actor interface{}, resource interface{}, allowWildcard bool) (map[interface{}]interface{}, error) {
+	results := make(map[interface{}]interface{})
+	query, err := (*o.p).queryRule("allow", actor, resource)
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		if v, err := query.Next(); err != nil {
+			return nil, err
+		} else if v == nil {
+			break
+		} else {
+			switch v := interface{}(v).(type) {
```

Comment:
The return type of `query.Next` is `*map[string]interface{}`

What you want to do is look for `v["action"]` (to match with the variable you pass in above).
And then on the result of that, you can apply this switch.

Reply:
Done ... I think. Compiles, so that's start.